redis/src/aio/connection.rs

```diff
@@ -378,10 +383,21 @@ async fn get_socket_addrs(host: &str, port: u16) -> RedisResult<SocketAddr> {
 pub(crate) async fn connect_simple<T: RedisRuntime>(
     connection_info: &ConnectionInfo,
 ) -> RedisResult<T> {
+    
     Ok(match connection_info.addr {
         ConnectionAddr::Tcp(ref host, port) => {
-            let socket_addr = get_socket_addrs(host, port).await?;
-            <T>::connect_tcp(socket_addr).await?
+            let socket_addrs = get_socket_addrs(host, port).await?;
```

Comment:
consider:
```rs
  let socket_addrs = get_socket_addrs(host, port).await?;
  for socket_addr in socket_addrs {
      let ret = <T>::connect_tcp(socket_addr).await;
      if ret.is_ok() {
          return ret;
      }
  }
 Err(RedisError::from((
      ErrorKind::InvalidClientConfig,
      "No address found for host",
  )))
```
or even simpler

```rs
match futures::future::select_ok(get_socket_addrs(host, port).await?.map(|socket_addr|<T>::connect_tcp(socket_addr)).await {
  Ok((res, _) => Ok(res),
  Err(_) =>  Err(RedisError::from((
                ErrorKind::InvalidClientConfig,
                "No address found for host",
            )))
}
```
(I hope these compile. Just wanted to provide suggestions)

Reply:
didn't notice that the dependencies include the `futures` package earlier :)